### ISSUE TITLE
Other minor auto-align quibbles

### ISSUE THREAD

**Issue by eevee** [sirjuddington/SLADE#486]
* [ ] There's no way to auto-align to the _left_, which I sometimes want to do to keep the right edge of a texture like `METAL2` aligned with a switch or doorframe or whatever.

* [ ] I'd love to be able to do a "justified" auto-align — given a contiguous set of lines, ensure the leftmost line starts at the left edge of the texture, and the rightmost line ends at the right edge of the texture.  If the total length isn't an exact multiple of the texture's width, spread the leftover space out evenly so there's no obvious seam.

    How good this looks depends heavily on the texture and architecture, but that basic algorithm is something I've done manually a good few times.  Imagine a marble column drawn with the shape tool whose perimeter ends up as being 200 — you can misalign by one pixel at a few vertices, and the obvious 8-pixel overlap all but disappears.

* [ ] Once I've done something like the above, I can't protect all my carefully-arranged offsets from being destroyed by an accidental auto-align.  :)

**Comment by Gaerzi** [comment]
In DB2 and GZDB, autoaligning is done relative to the highlighted sidedef. So if you have the cursor on the leftmost sidedef of a texture span, then all those to the right of it will be aligned to the left. SLADE 3 doesn't do that?

**Comment by eevee** [comment]
No, I think SLADE only aligns to the right.  You're right, though, it would make more sense to just align in both directions automatically.

In which case it might be nice to detect a loop of textures and automatically "justify".

Speaking of autoalign, I swear I've encountered an odd bug a couple times — I'll have a "cavey" room with the same texture all the way around (as confirmed by a shift-click), and I'll press Ctrl-A somewhere on it, but the alignment never makes it all the way around the room to affect the wall to the left.  I've never looked too much into it, though.
